src/Microsoft.AspNetCore.Sockets.Common/MessageBatchFormatter.cs

```diff
@@ -0,0 +1,134 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNetCore.Sockets
+{
+    public static class MessageBatchFormatter
+    {
+        private const byte BatchTerminator = (byte)';';
+        private const byte TextFormatIndicator = (byte)'T';
+        private const byte BinaryFormatIndicator = (byte)'B';
+
+        public static bool TryFormatMessages(IEnumerable<Message> messages, Span<byte> buffer, MessageFormat format, out int bytesWritten)
```

Comment:
Well, in general the desired goal is to write the entire batch in a single HTTP response when long-polling (to avoid round-trips to get more messages), so it would still have to make clear that it failed to write so that the caller could get more space for it to continue writing. Also, there'd need to be state-tracking to ensure that when resuming writing it doesn't write the prefix again.

I'm going to leave it this way for today and do some reviewing of how we're going to use it over the weekend. I definitely felt, as I was writing it, that it seemed odd to get 99% of the way through writing only to fail and discard it all.